File: app/src/main/java/com/cocopie/mobile/xgen/example/CoCoPIEJNIExporter.java
package com.cocopie.mobile.xgen.example;

public class CoCoPIEJNIExporter {
    static {
        try {
            System.loadLibrary("inference_api_jni");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static native long Create(String pbPath, String dataPath);

    public static native float[] Run(long engine, float[] input);
}

File: app/src/main/java/com/cocopie/mobile/xgen/example/CoCoPIEJNIExporter.kt
package com.cocopie.mobile.xgen.example

object CoCoPIEJNIExporter {

    init {
        try {
            System.loadLibrary("inference_api_jni")
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    external fun CreateOpt(pbPath: String?, dataPath: String?): Long
    external fun CreateFallback(fallbackPath: String?): Long
    external fun Run(engine: Long, input: FloatArray?): FloatArray?
}